interfaces/systemd/backend.go

```diff
@@ -78,18 +77,16 @@ func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repo
 			logger.Noticef("cannot reload systemd state: %s", err)
 		}
 	}
```

Comment:
As you mentioned, we need to `DisableNow` the removed units ahead of time. For that, let's please evaluate the glob above, and stop the files that are not found in `content`.